### PR TITLE
Adds a New Shirt Accessory

### PR DESCRIPTION
**Pull request by Geevies**
It's called Silversun because Human Lore. Ask Borya.

image
image

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/modules/clothing/under/accessories/shirts.dm [hunk 1]
```diff
@@ -52,4 +52,27 @@
...
+	open = !open
+	icon_state = "[initial(icon_state)][open ? "_open" : ""]"
+	item_state = icon_state
+	to_chat(user, span("notice", "You [open ? "open" : "close"] \the [src]."))
+	// the below forces the shirt to hard reset its image so it resets later its fucking weird ok
+	inv_overlay = null
+	mob_overlay = null
+
+/obj/item/clothing/accessory/silversun/red
+	desc = "A stylish Solarian shirt of Silversun design. This one is red."
```
Comment (alsoandanswer) [inline code/modules/clothing/under/accessories/shirts.dm]:
```suggestion
	desc = "A stylish Solarian shirt of Silversun design. This one is crimson."
```

File: code/modules/client/preference_setup/loadout/loadout_accessories.dm [hunk 2]
```diff
@@ -193,6 +193,17 @@
 	..()
 	gear_tweaks += gear_tweak_free_color_choice
 
+/datum/gear/accessory/silversun
+	display_name = "silversun shirt selection"
```
Comment (NonQueueingMatt) [inline code/modules/client/preference_setup/loadout/loadout_accessories.dm]: might wanna add that it's like a flower shirt or smth

File: code/modules/clothing/under/accessories/shirts.dm [hunk 3]
```diff
@@ -52,4 +52,27 @@
...
+	var/open = FALSE
+
+/obj/item/clothing/accessory/silversun/attack_self(mob/user)
+	open = !open
+	icon_state = "[initial(icon_state)][open ? "_open" : ""]"
+	item_state = icon_state
+	to_chat(user, span("notice", "You [open ? "open" : "close"] \the [src]."))
+	// the below forces the shirt to hard reset its image so it resets later its fucking weird ok
+	inv_overlay = null
+	mob_overlay = null
```
Comment (NonQueueingMatt) [inline code/modules/clothing/under/accessories/shirts.dm]: Why ain't this a verb? It'd be pretty weird to have to take your shirt off to unbutton it.

File: code/modules/clothing/under/accessories/shirts.dm [hunk 4]
```diff
@@ -52,4 +52,41 @@
...
+	set src in usr
+
+	if(!istype(usr, /mob/living))
+		return
+	if(use_check_and_message(usr))
+		return
+
+	if(ismob(usr))
+		var/mob/user = usr
+		attack_self(user)
```
Comment (NonQueueingMatt) [inline code/modules/clothing/under/accessories/shirts.dm]: You already checked if they're a mob or not before.

**Pull request merged**